### PR TITLE
Adding merge_commit_sha field to PullRequest model

### PR DESCRIPTION
**Pull request by jdesiloniz**
This PR adds the `merge_commit_sha` to the `PullRequest` model. According to the GitHub API docs:

```
The value of the merge_commit_sha attribute changes depending on the state of the pull request. Before merging a pull request, the merge_commit_sha attribute holds the SHA of the test merge commit. After merging a pull request, the merge_commit_sha attribute changes depending on how you merged the pull request:

If merged as a merge commit, merge_commit_sha represents the SHA of the merge commit.
If merged via a squash, merge_commit_sha represents the SHA of the squashed commit on the base branch.
If rebased, merge_commit_sha represents the commit that the base branch was updated to.
```

This field can be really useful for creating and updating statuses by targetting the test commit sha instead of the latest one in a PR.

### CODE REVIEW

**Comment by fedefernandez** [comment]
@jdesiloniz I guess we need to disable that option for scala 2.11, something like:

```scala
scalacOptions ++= (scalaBinaryVersion.value match {
  case x if x.startsWith("2.12") => Seq("-Ywarn-macros:after")
  case _                         => Seq.empty
})
```

File: project/ProjectPlugin.scala [hunk 1]
```diff
@@ -124,6 +124,10 @@ object ProjectPlugin extends AutoPlugin {
       scalaVersion := V.scala212,
       crossScalaVersions := Seq(V.scala211, V.scala212),
       scalacOptions ~= (_ filterNot Set("-Xlint").contains),
+      scalacOptions ++= ((scalaBinaryVersion.value, getEnvVar("SCALAENV")) match {
```
Comment (bilki) [inline project/ProjectPlugin.scala]: Is this still necessary?

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.20.2-SNAPSHOT"
\ No newline at end of file
+version in ThisBuild := "0.20.2"
```
Comment (bilki) [inline version.sbt]: Are we going to release now?

Comment (jdesiloniz) [reply version.sbt]: @bilki I'm holding back from this, so we can decide the release timeline appropiatedly in a separate PR.

File: project/ProjectPlugin.scala (hunk 1 above)
Comment (jdesiloniz) [reply project/ProjectPlugin.scala]: @bilki I've just launched a new build process in Travis without these, to verify if the stack size fix did the trick by itself... 🤔

Comment (jdesiloniz) [reply project/ProjectPlugin.scala]: @bilki yup, that did the trick! Thanks for the suggestion!!

**Pull request merged**